ui/litellm-dashboard/src/components/mcp_tools/create_mcp_server.tsx

```diff
@@ -0,0 +1,244 @@
+import { MCPServer } from "./types";
+import { isAdminRole } from "@/utils/roles";
+
+interface CreateMCPServerProps {
+  userRole: string;
+  accessToken: string | null;
+  onCreateSuccess: (newMcpServer: MCPServer) => void;
+}
+
+const CreateMCPServer: React.FC<CreateMCPServerProps> = ({
+  userRole,
+  accessToken,
+  onCreateSuccess,
+}) => {
+  const [form] = Form.useForm();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleCreate = async (formValues: Record<string, any>) => {
+    setIsLoading(true);
+    try {
+      console.log(`formValues: ${JSON.stringify(formValues)}`);
+
+      if (accessToken != null) {
+        const response: MCPServer = await createMCPServer(
+          accessToken,
+          formValues
+        );
+
+        message.success("MCP Server created successfully");
+        form.resetFields();
+        setModalVisible(false);
+        onCreateSuccess(response);
+      }
+    } catch (error) {
+      message.error("Error creating MCP Server: " + error, 20);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  // state
+  const [isModalVisible, setModalVisible] = useState(false);
+
+  const handleCancel = () => {
+    form.resetFields();
+    setModalVisible(false);
+  };
+
+  // rendering
+  if (!isAdminRole(userRole)) {
+    return null;
+  }
+
+  return (
+    <div>
+      <Button 
+        className="mx-auto" 
+        onClick={() => setModalVisible(true)}
+      >
+        + Create New MCP Server
+      </Button>
+
+      <Modal
+        title={
+          <div className="flex items-center space-x-3 pb-4 border-b border-gray-100">
+            <div className="w-10 h-10 bg-white rounded-lg flex items-center justify-center shadow-sm border border-gray-200">
+              <img 
+                src="/assets/logos/mcp_logo.png" 
+                alt="MCP Logo" 
+                className="w-8 h-8 object-contain"
+              />
+            </div>
+            <div>
+              <h2 className="text-xl font-semibold text-gray-900">Create New MCP Server</h2>
+              <p className="text-sm text-gray-500 mt-1">Configure your MCP server connection</p>
+            </div>
+          </div>
+        }
+        open={isModalVisible}
+        width={1000}
+        onCancel={handleCancel}
+        footer={null}
+        className="top-8"
+        styles={{
+          body: { padding: '24px' },
+          header: { padding: '24px 24px 0 24px', border: 'none' },
+        }}
+      >
+        <div className="mt-6">
+          <Form
+            form={form}
+            onFinish={handleCreate}
+            layout="vertical"
+            className="space-y-6"
+          >
+            <div className="grid grid-cols-1 gap-6">
+              <Form.Item
+                label={<span className="text-sm font-medium text-gray-700">MCP Server Name</span>}
+                name="alias"
+                rules={[
+                  { required: false, message: "Please enter a server name" },
```

Comment:
Can we add a tooltip or `?`.. something that says this is a best practice to include

Reply:
updated !